SabreTools.Models/PlayStation4/AppPkgHeader.cs

```diff
@@ -0,0 +1,238 @@
+        public byte[]? Zeroes1;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Entry 1
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainEntry1SHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Entry 2
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainEntry2SHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Digest Table
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? DigestTableSHA256;
+        
+        /// <summary>
+        /// PKG SHA256 for Main Table
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? MainTableSHA256;
+        
+        /// <summary>
+        /// PKG Padding Section 2
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 640)]
+        public byte[]? Zeroes2;
+
+        /// <summary>
+        /// PFS Unknown Field
+        /// </summary>
+        public uint PFSUnknown;
+
+        /// <summary>
+        /// PFS Image Count
+        /// </summary>
+        public uint PFSImageCount;
+
+        /// <summary>
+        /// PFS Image Flags
+        /// </summary>
+        public ulong PFSImageFlags;
+
+        /// <summary>
+        /// PFS Image Offset
+        /// </summary>
+        public ulong PFSImageOffset;
+
+        /// <summary>
+        /// PFS Image Size
+        /// </summary>
+        public ulong PFSImageSize;
+
+        /// <summary>
+        /// Mount Image Offset
+        /// </summary>
+        public ulong MountImageOffset;
+
+        /// <summary>
+        /// Mount Image Size
+        /// </summary>
+        public ulong MountImageSize;
+
+        /// <summary>
+        /// PKG Size
+        /// </summary>
+        public ulong PKGSize;
+
+        /// <summary>
+        /// PKG Signed Size
+        /// </summary>
+        public uint PKGSignedSize;
+
+        /// <summary>
+        /// PKG Signed Size
+        /// </summary>
+        public uint PKGCacheSize;
+        
+        /// <summary>
+        /// SHA256 for PFS Image
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? PFSImageSHA256;
+        
+        /// <summary>
+        /// SHA256 for PFS Signed
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 32)]
+        public byte[]? PFSSignedSHA256;
+
+        /// <summary>
+        /// PFS Split Size nth 0
+        /// </summary>
+        public ulong PFSSplitSize0;
+
+        /// <summary>
+        /// PFS Split Size nth 1
+        /// </summary>
+        public ulong PFSSplitSize1;
+        
+        /// <summary>
+        /// PKG Padding Section 3
+        /// </summary>
+        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 2896)]

```

Comment:
The offset from the end of `pfs_split_size_nth_1` to the start of `pkg_digest` appears to be 2900 (0xb54) and not 2896 (0xb50)

Reply:
There are 181 rows of 16 zero bytes, = 2896 bytes

Reply:
I'm not sure where that extra line is coming from, then. If you say the real data matches this, then okay